File: win-prog/ProgWinCS/Essential Structures/RandomClear/RandomClear.cs
//------------------------------------------
// RandomClear.cs © 2001 by Charles Petzold
//------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class RandomClear: Form 
{
     public static void Main()
     {
          Application.Run(new RandomClear());
     }
     public RandomClear()      
     {
          Text = "Random Clear";
     }
     protected override void OnPaint(PaintEventArgs pea)
     {
          Graphics grfx = pea.Graphics;
          Random   rand = new Random();

          grfx.Clear(Color.FromArgb(rand.Next(256),
                                    rand.Next(256),
                                    rand.Next(256)));
     }
}

File: win-prog/ProgWinVB/Essential Structures/RandomClear/RandomClear.vb
'--------------------------------------------
' RandomClear.vb (c) 2002 by Charles Petzold
'--------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class RandomClear
    Inherits Form

    Shared Sub Main()
        Application.Run(New RandomClear())
    End Sub

    Sub New()
        Text = "Random Clear"
    End Sub

    Protected Overrides Sub OnPaint(ByVal pea As PaintEventArgs)
        Dim grfx As Graphics = pea.Graphics
        Dim rand As New Random()

        grfx.Clear(Color.FromArgb(rand.Next(256), _
                                  rand.Next(256), _
                                  rand.Next(256)))
    End Sub
End Class
